### PR TITLE
feat: LTS release process

### PR DESCRIPTION
**Pull request by moolen**
This PR proposes a long-term support process and auto-update mechanism.

fixes: #2044

see #2160 and #2161 for generated PRs.

Proof of work:

* [fork] [build image from release-0.7 branch](https://github.com/moolen/external-secrets/actions/runs/4505796042/jobs/7931918993) after merge
* [fork] [create Release](https://github.com/moolen/external-secrets/actions/runs/4505534810/jobs/7932155465) Workflow for 0.7.3 (promoting release-0.7 image)

TODO:
- [x] run on `main` and `release-*` branches once per week
- [x] do an end to end test
- [x] check if we can exclude dependabot from caring about go.mod
- [x] add documentation around long-term support
- [x] create issue template for release task

After this PR got merged:
- [ ] update `release-0.8` and `release-0.7`
    - [ ] cherry-pick `.github` directory
    - [ ] backport `make update-deps`
    - [ ] protect release branches
- [ ] test release workflow 
    - [ ] release 0.7.x including #2180

### CODE REVIEW

File: design/006-LTS-release.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+WRT: https://github.com/external-secrets/external-secrets/issues/2044
+
+We want to provide security patches and critical bug fixes in a timely manner to our users.
+To do so, we offer long-term support for our latest two (N, N-1) software releases.
+We aim for a 2-3 month **minor** release cycle, i.e. a given release is supported for about 4-6 months.
+
+We want to cover the following cases:
+
+* weekly image rebuilds to update OS dependencies
+* weekly go dependency updates
```
Comment (moolen) [inline design/006-LTS-release.md:12]: dependency updates **must be approved** and merged by a maintainer. I.e. this is not fully automatic.

File: design/006-LTS-release.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    - [ ] version table
+    - [ ] Provider Stability and Support table
+    - [ ] Provider Feature Support table
+- [ ] docs: update [roadmap page](https://external-secrets.io/main/contributing/roadmap/)
+- [ ] tidy up [Project Board](https://github.com/orgs/external-secrets/projects/2)
+    - [ ] move issues to next milestone
+    - [ ] close milestone
+
+#### Release Execution
+- [ ] Follow the [Release Process guide](https://external-secrets.io/main/contributing/release/)
```
Comment (moolen) [inline design/006-LTS-release.md:47]: TODO: add instructions in regards to branch management to the contributing-release docs.

File: design/006-LTS-release.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+We want to provide security patches and critical bug fixes in a timely manner to our users.
+To do so, we offer long-term support for our latest two (N, N-1) software releases.
+We aim for a 2-3 month **minor** release cycle, i.e. a given release is supported for about 4-6 months.
+
+We want to cover the following cases:
+
+* weekly image rebuilds to update OS dependencies
+* weekly go dependency updates
+* backport bug fixes on demand
+
```
Comment (gusfcarvalho) [inline design/006-LTS-release.md:14]:
```suggestion

Note: features cut off on a minor release will not be backported to older releases.
```

File: design/006-LTS-release.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+## Automatic Updates
+
+We'll build a Github Action (GHA) which will automatically update the `go.mod` dependencies once per week or on request.
+The GHA will make the necessary code changes and opens a PR. Once approved and merged into `main` or `release-x.y` our build pipelines
+will build and push the artefact to ghcr.
+
```
Comment (gusfcarvalho) [inline design/006-LTS-release.md:20]:
```suggestion

## Manual Updates

Bug Fixes will be merged onto each release branch individually.
This is achieved by creating separate PRs from a corresponding branch of the release
(e.g. bug fixes targetting `release-1.0` should be created from `release-1.0` branch).
Once approved and merged into `main`  or `release-x.y`, ou build pipeline will build and push the artifact to ghcr
```

File: design/006-LTS-release.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+* weekly image rebuilds to update OS dependencies
+* weekly go dependency updates
+* backport bug fixes on demand
+
+## Automatic Updates
+
+We'll build a Github Action (GHA) which will automatically update the `go.mod` dependencies once per week or on request.
+The GHA will make the necessary code changes and opens a PR. Once approved and merged into `main` or `release-x.y` our build pipelines
+will build and push the artefact to ghcr.
```
Comment (gusfcarvalho) [inline design/006-LTS-release.md:19]:
```suggestion
will build and push the artifact to ghcr.
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - main
+      - release-*
```
Comment (moolen) [inline .github/workflows/ci.yml:7]: This should run for `release-*` branches as well to build an image which will be promoted by the "Create Release" workflow.

**Comment by knelasevero** [comment]
I say we go ahead. Sorry for delay

**Comment by knelasevero** [comment]
@sebagomez if you could also have a look. This one is important

**Comment by moolen** [comment]
@sebagomez can i get this in to do a release, any objections? :pray:

**Comment by sebagomez** [comment]
Sorry yes, I'm on with it, go ahead

**Pull request merged**